src/components/basic-element/basic-element.jsx

```diff
@@ -226,11 +223,15 @@ var BasicElement = React.createClass({
    * if they'd otherwise run off the page.
    */
   handleTranslation: function(x, y) {
-    var dl = this.dragLimits,
-        w_edge = dl.width,
-        h_edge = dl.height;
+    var rect_width = this.rect.width * this.state.scale;
```

Comment:
When finding the "actual"/visual width and height of the element after transforms, I'm not sure of the formula to take rotation into account -- @Pomax can you maybe help with this?